emacs-macport: use a different hammer for the `CF_NOESCAPE` bug

emilazy opened the pull request:
The header was breaking compilation with modern versions of Clang, seemingly due to the `limits.h` stuff. Wrapping the header directly is simpler and fixes it.

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

emilazy committed:
emacs-macport: use a different hammer for the `CF_NOESCAPE` bug

The header was breaking compilation with modern versions of Clang,
seemingly due to the `limits.h` stuff. Wrapping the header directly
is simpler and fixes it.

jian-lin reviewed (commented):
I am not familiar with `emacs-macport`.  Maybe @Atemu (#155360) and @tnytown (#252244) have something to say?

emilazy committed:
emacs-macport: fix build

I don’t love how this is separated from the actual declaration of
the source, but it seems to be what we have to do.

Fixes: 4a2ad8134b8b40f198c71f7d8d84467d250cd714

emilazy commented:
Hmm, I’m not getting that exact failure, but https://github.com/NixOS/nixpkgs/pull/437637 did regress the build. I’ve pushed a fix for that.

emilazy commented:
My test build passed. @kfiz, could you try it again and let me know exactly how you were testing?

I’ll let @jian-lin have another look given the new commit too.

Incidentally, I see that nobody has reported that `emacs-macport` currently fails to build on `nixpkgs-unstable` after 5 days. I’m wondering how popular the package is at this point.

kfiz commented:
> My test build passed. @kfiz, could you try it again and let me know exactly how you were testing?
> 
> I’ll let @jian-lin have another look given the new commit too.

@emilazy  works for me now.

kfiz commented:
> Incidentally, I see that nobody has reported that `emacs-macport` currently fails to build on `nixpkgs-unstable` after 5 days. I’m wondering how popular the package is at this point.

@emilazy  my strong assumption is that most of the `emacs-macport` users have temporarily switched to vanilla emacs to have a functional `emacs` `v30` version. They will switch back if there is a `v30` for `emacs-macport`.

jian-lin reviewed (commented):
The new commit "emacs-macport: fix build" looks good.

> I don’t love how this is separated from the actual declaration of
the source

`srcRepo` is separated to let downstream users be able to change it.

emilazy commented:
Thanks for the reviews :)

Maybe we should check whether @juuyokka is still interested in working on https://github.com/NixOS/nixpkgs/pull/393512, as it seems like the `emacs-macport` package may become a maintenance burden if it stays on an old version indefinitely.

kfiz commented:
> Maybe we should check whether @juuyokka is still interested in working on #393512, as it seems like the `emacs-macport` package may become a maintenance burden if it stays on an old version indefinitely.

@emilazy agree. I think in the meantime we should switch to the fork that is already mentioned there https://github.com/jdtsmith/emacs-mac until it is clear what happens to the original package. I've been test driving it and it seems quite stable to me. I can send a new PR that is based on #393512.
